### ISSUE TITLE
Error in custom API code

### ISSUE THREAD

**Issue by epogrebnyak** [mini-kep/db#70]
```[2017-11-14 08:56:10,428] ERROR in app: Exception on /ru/series/GDP/a/yoy/1998/2017 [GET]
2017-11-14T08:56:10.430891+00:00 app[web.1]: Traceback (most recent call last):
2017-11-14T08:56:10.430892+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.6/site-packages/flask/app.py", line 1982, in wsgi_app
2017-11-14T08:56:10.430893+00:00 app[web.1]:     response = self.full_dispatch_request()
2017-11-14T08:56:10.430897+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.6/site-packages/flask/app.py", line 1614, in full_dispatch_request
2017-11-14T08:56:10.430898+00:00 app[web.1]:     rv = self.handle_user_exception(e)
2017-11-14T08:56:10.430899+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.6/site-packages/flask/app.py", line 1517, in handle_user_exception
2017-11-14T08:56:10.430899+00:00 app[web.1]:     reraise(exc_type, exc_value, tb)
2017-11-14T08:56:10.430900+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.6/site-packages/flask/_compat.py", line 33, in reraise
2017-11-14T08:56:10.430901+00:00 app[web.1]:     raise value
2017-11-14T08:56:10.430902+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.6/site-packages/flask/app.py", line 1612, in full_dispatch_request
2017-11-14T08:56:10.430902+00:00 app[web.1]:     rv = self.dispatch_request()
2017-11-14T08:56:10.430903+00:00 app[web.1]:   File "/app/.heroku/python/lib/python3.6/site-packages/flask/app.py", line 1598, in dispatch_request
2017-11-14T08:56:10.430904+00:00 app[web.1]:     return self.view_functions[rule.endpoint](**req.view_args)
2017-11-14T08:56:10.430905+00:00 app[web.1]:   File "/app/db/custom_api/views.py", line 22, in time_series_api_interface
2017-11-14T08:56:10.430906+00:00 app[web.1]:     data = DatapointOperations.select(**this_variable.query_param)
2017-11-14T08:56:10.430906+00:00 app[web.1]:   File "/app/db/custom_api/decomposer.py", line 202, in query_param
2017-11-14T08:56:10.430907+00:00 app[web.1]:     return {key: getattr(self, key) for key in keys}
2017-11-14T08:56:10.430908+00:00 app[web.1]:   File "/app/db/custom_api/decomposer.py", line 202, in <dictcomp>
2017-11-14T08:56:10.430908+00:00 app[web.1]:     return {key: getattr(self, key) for key in keys}
2017-11-14T08:56:10.430909+00:00 app[web.1]:   File "/app/db/custom_api/decomposer.py", line 183, in end_date
2017-11-14T08:56:10.430910+00:00 app[web.1]:     return self.token.end
2017-11-14T08:56:10.430910+00:00 app[web.1]:   File "/app/db/custom_api/decomposer.py", line 139, in end
2017-11-14T08:56:10.430911+00:00 app[web.1]:     return as_date(end_year, month=12, day=31)
2017-11-14T08:56:10.431045+00:00 app[web.1]: NameError: name 'end_year' is not defined```

**Comment by epogrebnyak** [comment]
This happens on <http://minikep-db.herokuapp.com/ru/series/GDP/a/yoy/1998/2017>, and it is not tested.

**Comment by epogrebnyak** [comment]
Closed by 9392143df2e106bc8ad1e92d067d83ef6767926e
